### ISSUE TITLE
TTlock handle in Mirror issue

### ISSUE THREAD

**Issue by davemccollough-geokey** [ttlock/iOS_SDK_Demo#26]
https://github.com/ttlock/iOS_SDK_Demo/assets/98485073/c570ebb4-3c47-4584-8358-93ff4a7c8e6a

Please check this video. If we press 1 it got pressed 6, If we press 2 it got pressed 7.  We have also tried after hard resetting a lock but issue is still persist.

Please provide solution.

**Comment by ttlock** [comment]
1 Please send the lock Name to us. Let's locate the issue.
2 Do you have WeChat? If have, please post it. We have a developer group.

**Comment by ttlock** [comment]
Please send the lock Name to us. Let's locate the issue.

**Comment by JackeyPSSPL** [comment]
@ttlock 
We have getting issue in below locks

(A)
"lockAlias": "MyTestLock2024/01/17 13:37:28"
"lockMac": "1F:2A:B3:74:C9:EC"
"lockId": 7254378

(B)
"lockAlias": "MyTestLock2024/01/18 14:29:15"
"lockMac": "EA:19:91:7A:47:A2"
"lockId": 6635540

(C)
"lockAlias": "MyTestLock2024/01/18 14:28:49"
"lockMac": "5B:6B:33:F4:CE:06"
"lockId": 6569398

also lock and unlock operation have opposite 

I wanted to bring to your attention a clarification regarding the lock and unlock operations with TTLock. It has come to my notice that the system follows an opposite mechanism for these actions.
 In other words, if an attempt is made to lock, the system responds with an unlock, and conversely, when attempting to unlock, the system initiates a lock. This behavior may seem counterintuitive

**Comment by ttlock** [comment]
ok, we will get back to you as soon as possible.

**Comment by ttlock** [comment]
Have we configured firmware upgrades for these locks ?
It's because the current firmware version doesn't match the lock.
This issue should be resolved by rolling back the firmware version.
We will check it next working day

**Comment by ttlock** [comment]
Please try firmware upgrade again

**Comment by JackeyPSSPL** [comment]
we are getting in two locks below response
locks are:

(B) EA:19:91:7A:47:A2
(C) 5B:6B:33:F4:CE:06

so it's means no upgrade available

```
{
        "needUpgrade": 0,
        "modelNum": "SN373-150MS-T20-INT_PV53",
        "hardwareRevision": "1.1.1",
        "firmwareInfo": {
            "modelNum": "SN373-150MS-T20-INT_PV53",
            "hardwareRevision": "1.1.1",
            "firmwareRevision": "6.4.11.220206"
        },
        "firmwarePackage": "",
        "version": "",
        "firmwareRevision": "6.4.11.220206"
    }
```

and in remaining lock ttlock details API is not give any response so not able to identify which firmware version currently we have.
lock is
(A) 1F:2A:B3:74:C9:EC

So we are sharing that account which is added these 3 locks so if possible from your side then do upgrade from your side and let us know for same.

 **Username: jackey.kabra@geokeyaccess.com
 Password: Admin@123**
 
 Thanks to support team to give a quick a reply.

**Comment by ttlock** [comment]
It is impossible from our side to upgrade your locks because it needs to connect locks.
We have updated, please try  firmware upgrade again.

**Comment by JackeyPSSPL** [comment]
@ttlock Thank you of your support Now it's working as we accepted.
